### DISCUSSION TITLE
Fixed number of up or down spin ansatz circuit

### DISCUSSION THREAD

**Discussion by biswajitpaul18** [jcmgray/quimb#173] [Q&A] [answered]
In the example, https://quimb.readthedocs.io/en/latest/examples/ex_tn_train_circuit.html# the number of up or down spins is not conserved. Can you give me some idea on how can I construct an ansatz circuit where the number of up or down spins will be conserved? or can you provide some example code regarding the same?

**Comment by jcmgray** [comment, marked as answer]
Hi @biswajitpaul18, I don't know exactly what is done in research papers, but one way is to simply choose gates which locally preserve number, e.g. FSIM / FSIMG, CZ, RZZ, RZ, phase Cphase, etc.

**Comment by biswajitpaul18** [comment]
Thanks, @jcmgray for your reply. I was using the CNOT gate that's why the number of up or down spins was not conserved. Now I am using the CZ gate instead of the CNOT gate, it works now I am getting the conserved number of up or down spins at the output.

thanks a lot for your help.
